### ISSUE TITLE
HTML5: Gamepads without mapping should fallback to standard mapping

### ISSUE THREAD

**Issue by aglitchman** [defold/defold#7040]
**Is your feature request related to a problem? Please describe (REQUIRED):**
You know that there are lots of different gamepads on the market and it's hard to collect and maintain all possible mappings for them. Luckily, web browsers already do that hard work for us because Gamepad API has [a standard mapping](https://w3c.github.io/gamepad/#remapping) that looks very much like DualShock controller:

![image](https://user-images.githubusercontent.com/193258/194433275-c187d461-dbaa-4ab5-bda3-6f6ea19bb279.png)

More info - https://developer.mozilla.org/en-US/docs/Web/API/Gamepad_API/Using_the_Gamepad_API

> `mapping` - A string indicating whether the browser has remapped the controls on the device to a known layout. Currently there is only one supported known layout — the [standard gamepad](https://w3c.github.io/gamepad/gamepad.html#remapping). If the browser is able to map controls on the device to that layout the mapping property will be set to the string `standard`.

To sum up, it would be easier for developers if Defold was able to get input from any _**standard**_ gamepad.

**Describe the solution you'd like (REQUIRED):**
Defold should fallback to standard mapping of Gamepad API if the `mapping` field of a gamepad equals `standard` and there is no mapping for that controller in the `default.gamepads` file.

**Additional context (OPTIONAL):**
For example, I attached NVIDIA Shield Controller (this is quite a rare model) to my PC some time ago; it had the "standard" mapping and I was able to use it in the most web games (except games made with Defold).

**Comment by aglitchman** [comment]
> For example, I attached NVIDIA Shield Controller (this is quite a rare model) to my PC some time ago; it had the "standard" mapping and I was able to use it in the most web games (except games made with Defold).

![](https://user-images.githubusercontent.com/193258/194621116-79c427c7-42a0-49ab-adc7-c5b938d5c453.png)

**Comment by britzl** [comment]
This is an interesting suggestion! I'll definitely look into this!
